nucleus/common/common-util/src/main/java/com/sun/enterprise/loader/ASURLClassLoader.java

```diff
@@ -795,21 +792,16 @@ protected synchronized ClassData findClassData(String name) throws ClassNotFound
      * @throws IOException if an i/o error
      */
     private byte[] getClassData(InputStream istream) throws IOException {
-        BufferedInputStream bstream = new BufferedInputStream(istream);
         byte[] buf = new byte[4096];
         ByteArrayOutputStream bout = new ByteArrayOutputStream();
         int num = 0;
-        try {
+        try(BufferedInputStream bstream = new BufferedInputStream(istream)) {
             while( (num = bstream.read(buf)) != -1) {
                 bout.write(buf, 0, num);
             }
-        } finally {
-            try {
-                bstream.close();
-            } catch (IOException closeIOE) {
-                ASURLClassLoader._logger.log(INFO,
-                        CULoggerInfo.exceptionInASURLClassLoader, closeIOE);
-            }
+        } catch (IOException closeIOE) {
```

Comment:
This doesn't catch just close IOE, but all IOExceptions in the try. Perhaps just don't catch it, it is always a failure, same if bstream.read fails and if we cannot close it.

Reply:
Fixed.